agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/sampling/SamplingOverridesTest.java

```diff
@@ -44,157 +46,243 @@ void shouldSampleByDefault() {
     Attributes attributes = Attributes.empty();
 
     // expect
-    assertThat(sampler.getOverride(attributes)).isNull();
+    assertThat(sampler.getOverride(SpanKind.SERVER, attributes)).isNull();
+  }
+
+  @Test
+  void shouldFilterBySpanKind() {
+    // given
+    List<SamplingOverride> overrides = singletonList(newOverride(Configuration.SpanKind.SERVER, 0));
+    SamplingOverrides sampler = new SamplingOverrides(overrides);
+    Attributes attributes = Attributes.empty();
+
+    // expect
+    assertThat(sampler.getOverride(SpanKind.SERVER, attributes)).isNotNull();
+  }
+
+  @Test
+  void shouldNotFilterBySpanKind() {
+    // given
+    List<SamplingOverride> overrides = singletonList(newOverride(Configuration.SpanKind.SERVER, 0));
+    SamplingOverrides sampler = new SamplingOverrides(overrides);
+    Attributes attributes = Attributes.empty();
+
+    // expect
+    assertThat(sampler.getOverride(SpanKind.CLIENT, attributes)).isNull();
   }
 
   @Test
   void shouldFilterStrictMatch() {
     // given
     List<SamplingOverride> overrides =
-        singletonList(newOverride(0, newStrictAttribute("one", "1")));
+        singletonList(
+            newOverride(Configuration.SpanKind.SERVER, 0, newStrictAttribute("one", "1")));
+    SamplingOverrides sampler = new SamplingOverrides(overrides);
+    Attributes attributes = Attributes.of(AttributeKey.stringKey("one"), "1");
+
+    // expect
+    assertThat(sampler.getOverride(SpanKind.SERVER, attributes).getPercentage()).isEqualTo(0);
+  }
+
+  @Test
+  void shouldFilterStrictMatchWithNullSpanKind() {
+    // given
+    List<SamplingOverride> overrides =
+        singletonList(newOverride(null, 0, newStrictAttribute("one", "1")));
```

Comment:
when it's null, it will match any span kind?

Reply:
yes